### ISSUE TITLE
Support: Using MISP platform including MISP core and MISP objects without any modification and Wrapping them using REST Interface that my company makes.

### ISSUE THREAD

**Issue by r3search3r** [MISP/MISP#8095]
### Support Questions

Dear MISP project.

I'm writing this email to you because I have a question about violating AGPL v3 by using MISP platform without any modification.

1. If I use it for MISP platform including MISP core and MISP objects without any modifications to the code, doesn't it violate AGPL v3?

2. If it's possible, in case of servicing my RESTful API interface after wrapping the MISP platform, does it also violates AGPL v3?

Please feel free to contact us if you have any questions.

Best regards.

### MISP version

2.4.135

### Operating System

Ubuntu

### Operating System version

20.04

### PHP version

7.3

### Relevant log output

```shell
N/A
```

### Extra attachments

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by adulau** [comment]
Hi @r3search3r,  

Thank you for the very good questions. 

1- The AGPL only applies if there is a distribution and modification of the code as we mentioned in [the website](https://www.misp-project.org/license/#agpl-affero-general-public-license---misp-point-of-view). For your first question, there is no requirement of publication, if you run MISP without any modification. 

2- Same for the ReST API (any web services), if you don't modify the software, there is no need to publish it. 

To summarise, the AGPL (which is just a GPL with an additional clause) `if you run a modified program on a server and let other users communicate with it there, your server must also allow them to download the source code corresponding to the modified version running there.`

Let me know if it's clear for you.

Kind regards

**Comment by r3search3r** [comment]
Huge thanks to you about your quick and concise answer 👍 

> 1- The AGPL only applies if there is a distribution and modification of the code as we mentioned in the website. For your first question, there is no requirement of publication, if you run MISP without any modification.
>
> 2- Same for the ReST API (any web services), if you don't modify the software, there is no need to publish it.

You already announced me about the case of violation AGPL v3, I have a question about using words of _"publish"_. Does it means that If I modify any MISP core and MISP objects, I have to _"publish"_ to download all source related to MISP core and MISP objects?

Sincerely yours.

**Comment by adulau** [comment]
You're welcome.

If you do modification of the MISP core software and provide access to third-parties (private modification are still possible), you need to release (publish) the modified source to the users of the modified MISP application. That's why we always recommend to propose a pull-request to the MISP core software, you'll help the community and maintain your MISP software as not modified.

MISP objects are [dual-licensed](https://github.com/MISP/misp-objects/blob/main/LICENSE.md) under a CC-0 license (public domain) or simplified 2-clause BSD. So the licensing of the objects are super lax and can be freely reused in many use-cases without any need of publication.

Cheers

**Comment by r3search3r** [comment]
I understood, really appreciate it. :smile:

If I have more questions about this issue, Do I make another issue in this repository and link this issue?

**Comment by adulau** [comment]
Sure feel free. Such discussion can be useful for other members of the MISP community, If you think something [need clarification on the website about the open source license](https://www.misp-project.org/license/), let us know. Thank you too.
